Removed unused Classes

GabrielBrascher opened the pull request:
Remove unused Classes (classes with no references:
- com.cloud.agent.api.CheckStateAnswer
- com.cloud.agent.api.StartupVMMAgentCommand
- com.cloud.agent.api.routing.UserDataCommand
	- remove from description at
com.cloud.configuration.Config.ExecuteInSequenceNetworkElementCommands
enum
- com.cloud.agent.api.storage.UpgradeDiskCommand
- com.cloud.agent.api.storage.CreatePrivateTemplateCommand
- com.cloud.agent.api.storage.DestroyAnswer
	- Note: there is just a line of `//FIXME: Should have an DestroyAnswer` at
com.cloud.storage.resource.StoragePoolResource
- com.cloud.agent.api.storage.UpgradeDiskAnswer
- com.cloud.agent.api.storage.ManageVolumeAvailabilityAnswer
- com.cloud.agent.api.storage.ManageVolumeAvailabilityCommand
- com.cloud.exception.UsageServerException
- com.cloud.info.SecStorageVmLoadInfo
- com.cloud.serializer.SerializerHelper

DaanHoogland commented:
@GabrielBrascher removing unused code is good; less is more. These are API calls and related stuff however. A large test set should be run and I am surprised no test code would be touched by this change.

GabrielBrascher commented:
@DaanHoogland the lack of test in some classes indeed is a problem. However, these classes that I removed are not being used.

I used UCDetector (http://www.ucdetector.org/) as a plugin for Eclipse.  With this tool, I discovered a lot of code without any reference (variables, methods and classes).

This pull request had the goal of removing some of these classes. To ensure that I wasn't missing anything I searched for any file that could reference some answer or API command. As I haven't found any way of these classes being used, I removed them.

Any of those API commands that I deleted are not using org.apache.cloudstack.api.APICommand (with `@APICommand`). Therefore, those commands are not exposed in the ACS API.

DaanHoogland commented:
Ok, that is somewhat comforting, however still a large integration test suite should be run on this.

swill commented:
With these major stripping out of unused code, we need to be very careful that we do not introduce integration test issues.  Can you please validate that the integration test output does not change based on this code being removed?

GabrielBrascher commented:
@DaanHoogland @swill thank you for the concern. I agree with the need for executing integration/functional tests to keep the code stable. Unfortunately, our test environment is not set yet and it may take a while. As soon as we have it operational I will run the tests.

swill commented:
Thank you @GabrielBrascher.  I am also getting my CI setup so I can get tests running against these PRs, so I should be able to start validating this stuff as well.  Thanks for the continued effort.  :)

bvbharatk commented:
### ACS CI BVT Run
 **Sumarry:**
 Build Number 155
 Hypervisor xenserver
 NetworkType Advanced
 Passed=68
 Failed=5
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_vpc_vpn.py

 * ContextSuite context=TestRVPCSite2SiteVpn>:setup Failing since 14 runs

 * ContextSuite context=TestVpcRemoteAccessVpn>:setup Failing since 14 runs

 * ContextSuite context=TestVpcSite2SiteVpn>:setup Failing since 14 runs

* test_volumes.py

 * test_06_download_detached_volume Failed

* test_vm_life_cycle.py

 * test_10_attachAndDetach_iso Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_disk_offerings.py

rhtyd commented:
@GabrielBrascher can you rebase against latest master

GabrielBrascher commented:
Rebased against latest master. Thanks @rhtyd.

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1063

GabrielBrascher commented:
@rhtyd can you please run integration tests? Thanks ;)

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1112

rhtyd commented on server/src/com/cloud/configuration/Config.java:
```diff
@@ -1900,9 +1900,9 @@
-            "If set to true, DhcpEntryCommand, SavePasswordCommand, UserDataCommand, VmDataCommand will be synchronized on the agent side."
-                + " If set to false, these commands become asynchronous. Default value is false.",
-            null),
+                    "If set to true, DhcpEntryCommand, SavePasswordCommand, VmDataCommand will be synchronized on the agent side."
```
@GabrielBrascher can you fix indentations changes in this class/file?

GabrielBrascher replied:
Fixed, thanks @rhtyd.

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
@blueorangutan package

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1413

rafaelweingartner commented:
@blueorangutan package

@rhtyd I took the liberty of getting this PR and fixing the conflicts. There were quite a few unused classes that we removed.

blueorangutan commented:
@rafaelweingartner a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2351

nvazquez commented on server/src/main/java/com/cloud/configuration/Config.java:
```diff
@@ -1836,9 +1836,9 @@
             Boolean.class,
             "execute.in.sequence.network.element.commands",
             "false",
-            "If set to true, DhcpEntryCommand, SavePasswordCommand, UserDataCommand, VmDataCommand will be synchronized on the agent side."
```
Also found some reference on the `schema-410to420.sql` file

rafaelweingartner replied:
Good catch. I will fix that, and create a script to update the database.

nvazquez commented:
@blueorangutan test

blueorangutan commented:
@nvazquez a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

nvazquez commented:
@rafaelweingartner was a failure downloading dependencies before environment has been built, let me start them again
@blueorangutan test

blueorangutan commented:
@nvazquez a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3085)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 22147 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr1448-t3085-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Smoke tests completed. 67 look OK, 2 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 22.09 | test_multipleips_per_nic.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 469.24 | test_vpc_redundant.py

rafaelweingartner commented:
@rhtyd can I consider these tests results as successful?

rhtyd commented on engine/schema/src/main/resources/META-INF/db/schema-41120to41200.sql:
```diff
@@ -34,4 +34,7 @@ INSERT INTO `cloud`.`role_permissions` (`uuid`, `role_id`, `rule`, `permission`,
+UPDATE `cloud`.`async_job` SET `removed` = now() WHERE `removed` IS NULL;
+
+-- PR#1448 update description of 'execute.in.sequence.network.element.commands' parameter to reflect an unused command that has been removed. The removed class command is 'UserDataCommand'.
+update configuration set description = 'If set to true, DhcpEntryCommand, SavePasswordCommand, VmDataCommand will be synchronized on the agent side. If set to false, these commands become asynchronous. Default value is false.' where name = 'execute.in.sequence.network.element.commands'; 
```
@rafaelweingartner not the comment is to rewrite the update statement as `UPDATE `cloud`.`configuration` SET ...`

rafaelweingartner replied:
Ah, sure the schema. Done!
